### DISCUSSION TITLE
Duration Only Mode gone in v2.x?

### DISCUSSION THREAD

**Discussion by tofuSCHNITZEL** [kimai/kimai#3724] [General]
Hi,

I saw that DurationOnlyMode (Timesheet\TrackingMode) is to be removed in version 2.1?

will there be a replacement for it, e.g. an option for displaying a duration for "DefaultMode"?

cheers

**Comment by kevinpapst** [comment]
I don't understand the question. The "duration only" mode was always a "broken idea".
It removed the end field, which has no benefit, besides a "cleaner ui"
But this cleaner UI comes with an increased cognitive load (AKA worse UX), because you do not see the end time, but have to hope that you calculated it correctly.

<img width="717" alt="Bildschirm­foto 2023-01-03 um 16 45 38" src="https://user-images.githubusercontent.com/533162/210391473-95816876-370a-4fb7-84a7-f9961b7b4286.png">

The new screen has the best of both worlds: you select a duration and boom => the end is auto-calculated.
This is much cleaner IMO.
What are you missing here? What is your question exactly?

**Reply by tofuSCHNITZEL** [reply]
okay I see - sorry I tought the possibility to enter a duration will be gone. I have never used the normal mode so I overlooked that in the normal mode there is also a duration field...

I still "dream" about the possibility to enter a negative duration so I can enter times after I did the work and not have to calculate the start time... I noticed that recording the time after I did the work fits 99% of my workflow - I never know when customers call and therfore can start the time recording when the call starts but rather after finishing the call I will go to the timetracker and enter 15m as a duration and then have to correct the start time back manually back.

I'm currently looking through the DemoBundle and other extensions to maybe create an extension of my own that would allow this - I'm also not new to PHP but never worked with this framwork and I can't really find which file does what and where to start with an extension of my own...

I would gladly sponsor this feature or extension and also would want it to be available to everybody afterwards... (the time it would cost me to do it, I would rather pay to Keleo ;))

**Comment by kevinpapst** [comment]
See this:

https://user-images.githubusercontent.com/533162/210395257-750347f1-5581-4605-a1c1-d73511a7fafe.mov

You could try to work on a patch in this file, supporting "negative durations":
https://github.com/kimai/kimai/blob/main/assets/js/forms/KimaiTimesheetForm.js#L286

What I would imagine is: if a negative duration is detected, then the end time - duration => start time. If the end time is not yet set, then it is set to "now".

**Reply by tofuSCHNITZEL** [reply]
> What I would imagine is: if a negative duration is detected, then the end time - duration => start time. If the end time is not yet set, then it is set to "now".

I will implement this and submit a pull request. Thanks!!

**Comment by tofuSCHNITZEL** [comment]
sorry for hijacking this discussion.
I just noticed that probably not even a backend change is needed... just the JS that fills the end time needs to be able to handle the negative duration...
thanks for pointing me in the right direction!
